include/usb/class/usb_audio.h

```diff
@@ -0,0 +1,286 @@
+	USB_AUDIO_FU_DELAY_CONTROL		= 0x08,
+	USB_AUDIO_FU_BASS_BOOST_CONTROL		= 0x09,
+	USB_AUDIO_FU_LOUDNESS_CONTROL		= 0x0A
+};
+
+/** USB Terminal Types
+ * Refer to Table 2-1 - Table 2-4 from termt10.pdf
+ */
+enum usb_audio_terminal_types {
+	/* USB Terminal Types */
+	USB_AUDIO_USB_UNDEFINED   = 0x0100,
+	USB_AUDIO_USB_STREAMING   = 0x0101,
+	USB_AUDIO_USB_VENDOR_SPEC = 0x01FF,
+
+	/* Input Terminal Types */
+	USB_AUDIO_IN_UNDEFINED      = 0x0200,
+	USB_AUDIO_IN_MICROPHONE     = 0x0201,
+	USB_AUDIO_IN_DESKTOP_MIC    = 0x0202,
+	USB_AUDIO_IN_PERSONAL_MIC   = 0x0203,
+	USB_AUDIO_IN_OM_DIR_MIC     = 0x0204,
+	USB_AUDIO_IN_MIC_ARRAY      = 0x0205,
+	USB_AUDIO_IN_PROC_MIC_ARRAY = 0x0205,
+
+	/* Output Terminal Types */
+	USB_AUDIO_OUT_UNDEFINED		= 0x0300,
+	USB_AUDIO_OUT_SPEAKER		= 0x0301,
+	USB_AUDIO_OUT_HEADPHONES	= 0x0302,
+	USB_AUDIO_OUT_HEAD_AUDIO	= 0x0303,
+	USB_AUDIO_OUT_DESKTOP_SPEAKER	= 0x0304,
+	USB_AUDIO_OUT_ROOM_SPEAKER	= 0x0305,
+	USB_AUDIO_OUT_COMM_SPEAKER	= 0x0306,
+	USB_AUDIO_OUT_LOW_FREQ_SPEAKER	= 0x0307,
+
+	/* Bi-directional Terminal Types */
+	USB_AUDIO_IO_UNDEFINED			= 0x0400,
+	USB_AUDIO_IO_HANDSET			= 0x0401,
+	USB_AUDIO_IO_HEADSET			= 0x0402,
+	USB_AUDIO_IO_SPEAKERPHONE_ECHO_NONE	= 0x0403,
+	USB_AUDIO_IO_SPEAKERPHONE_ECHO_SUP	= 0x0404,
+	USB_AUDIO_IO_SPEAKERPHONE_ECHO_CAN	= 0x0405,
+};
+
+enum usb_audio_direction {
+	USB_AUDIO_IN = 0x00,
+	USB_AUDIO_OUT = 0x01
+};
+
+struct usb_audio_feature_unit_evt {
+	struct device *dev;
+	enum usb_audio_direction dir;
+	enum usb_audio_feature_unit_control_selectors cs;
+	u8_t channel;
+	void *val;
+};
+
+/**
+ * @brief Callback type used to inform the app that data were requested
+ *	  from the device and may be send to the Host.
+ *	  For sending the data usb_audio_send() API function should be used.
+ *
+ * @note User may not use this callback and may try to send in 1ms task
+ *	 instead. Sending every 1ms may be unsuccessful and may return -EAGAIN
+ *	 if Host did not required data.
+ *
+ * @param dev The device for which data were requested by the Host.
+ */
+typedef void (*usb_audio_data_request_cb_t)(const struct device *dev);
+
+/**
+ * @brief Callback type used to inform the app that data were successfully
+ *	  send/received.
+ *
+ * @param dev	 The device for which the callback was called.
+ * @param buffer Pointer to the net_buf data chunk that was successfully
+ *		 send/received.
+ * @param size	 Amount of data that were successfully send/received.
+ */
+typedef void (*usb_audio_data_completion_cb_t)(const struct device *dev,
+					       struct net_buf *buffer,
+					       size_t size);
+
+/**
+ * @brief Callback type used to inform the app that Host has changed
+ *	  one of the features configured for the device.
+ *	  Applicable for all devices.
+ *
+ * @warning Host may not use all of configured features.
+ *
+ * @param evt Event information to be parsed by app.
+ */
+typedef void (*usb_audio_feature_updated_cb_t)(
+					struct usb_audio_feature_unit_evt evt);
```

Comment:
For consistency, it would be better to pass the `dev` pointer in the first parameter of the callback function, not as a field of this structure.